src/components/SelectableMainTable.tsx

```diff
@@ -0,0 +1,192 @@
+            toggleClassName="has-icon u-no-margin--bottom"
+            toggleLabel={<Icon name="chevron-down" />}
+            toggleProps={{
+              "aria-label": "multiselect rows",
+            }}
+            links={[
+              {
+                children: `Select all ${itemName}s on this page`,
+                onClick: selectPage,
+              },
+              {
+                children: `Select all ${parentName} ${itemName}s`,
+                onClick: selectAll,
+              },
+            ]}
+          />
+        </>
+      ),
+      className: "select select-header",
+    },
+    ...(headers ?? []),
+  ];
+
+  const rowsWithCheckbox = rows.map((row) => {
+    const isRowSelected = selectedNames.includes(row.name ?? "");
+    const isRowProcessing = processingNames.includes(row.name ?? "");
+
+    const toggleRow = () => {
+      const newSelection = isRowSelected
+        ? selectedNames.filter((candidate) => candidate !== row.name)
+        : [...selectedNames, row.name ?? ""];
+      setSelectedNames(newSelection);
+    };
+
+    row.columns = [
+      {
+        content: (
+          <CheckboxInput
+            label=""
+            labelClassName="u-no-margin--bottom"
+            checked={isRowSelected}
+            onChange={toggleRow}
+            disabled={isRowProcessing}
+            aria-label={`Select ${row.name ?? ""}`}
+          />
+        ),
+        role: "rowheader",
+        className: "select",
+      },
+      ...(row.columns ?? []),
+    ];
+
+    row.className = classnames(row.className, {
+      "selected-row": isRowSelected,
+      "processing-row": isRowProcessing,
+    });
+    row.key = row.name;
+
+    return row;
+  });
+
+  const selectionState = isSomeSelected
+    ? [
+        {
+          className: "select-notification",
+          key: "select-info",
+          expanded: true,
+          expandedContent: (
+            <Notification
+              borderless
+              className="u-no-margin--bottom"
+              title="Selection"
+            >
+              {isAllSelected ? (
+                <>
+                  {allNames.length === 1 ? (
+                    <>
+                      <b>1</b> {itemName} of this {parentName} is selected.{" "}
+                    </>
+                  ) : (
+                    <>
+                      All <b>{allNames.length}</b> {itemName}s of this{" "}
+                      {parentName} are selected.{" "}
+                    </>
+                  )}
+                  <Button
+                    appearance="link"
+                    className="u-no-margin--bottom u-no-padding--top"
+                    onClick={selectNone}
+                  >
+                    Clear selection
+                  </Button>
+                </>
+              ) : (
+                <>
+                  <b>{selectedNames.length}</b> {itemName}
+                  {selectedNames.length > 1 && "s"} selected{" "}
```

Comment:
```suggestion
                  {selectedNames.length > 1 && "s"} selected.{" "}
```
Sorry, I just happened to notice this now - in the `isAllSelected` case above, we have a "." at the end of the message, but here we don't. I suggest adding it here too for consistency - and also because it looks better, given that we have another sentence right next to it.